src/index.tsx

```diff
@@ -1,10 +1,29 @@
 import ReactDOM from 'react-dom';
-import { HashRouter } from 'react-router-dom';
+import {
+  HashRouter,
+  Navigate,
+  Route,
+  Routes,
+} from 'react-router-dom';
 import { App } from './App';
+import { HomePage, NotFoundPage, TabsPage } from './pages';
 
 ReactDOM.render(
   <HashRouter>
-    <App />
+    <Routes>
+      <Route path="/" element={<App />}>
+        <Route index element={<HomePage />} />
+        <Route path="home" element={<Navigate to="/" />} />
+        <Route path="tabs">
+          <Route index element={<TabsPage />} />
+          <Route path=":tabId" element={<TabsPage />} />
+        </Route>
+        <Route
+          path="*"
+          element={<NotFoundPage />}
+        />
+      </Route>
+    </Routes>
```

Comment:
Better to have these Routes inside `App` component and create `Layout` component for '/' Route element